### ISSUE TITLE
Algorithm negotiation fail ssh-rsa

### ISSUE THREAD

**Issue by tklinchik** [robtimus/sftp-fs#12]
Hi,

trying to use your library and ran into issue upon initial connection:
"Algorithm negotiation fail: algorithmName="server_host_key" jschProposal="rsa-sha2-512,rsa-sha2-256,ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521" serverProposal="ssh-rsa""

I'm using:

```
    .withUsername("...")
    .withUserInfo(SimpleUserInfo("...").toCharArray())
    .withKnownHosts(file)
```

Would appreciate some guidance on how to make this work.

Here is a full stack:
```
Exception in thread "main" java.nio.file.FileSystemException: Algorithm negotiation fail: algorithmName="server_host_key" jschProposal="rsa-sha2-512,rsa-sha2-256,ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521" serverProposal="ssh-rsa"
	at com.github.robtimus.filesystems.sftp.SFTPEnvironment.asFileSystemException(SFTPEnvironment.java:730)
	at com.github.robtimus.filesystems.sftp.SFTPEnvironment.connect(SFTPEnvironment.java:655)
	at com.github.robtimus.filesystems.sftp.SFTPEnvironment.openChannel(SFTPEnvironment.java:526)
	at com.github.robtimus.filesystems.sftp.SSHChannelPool$Channel.<init>(SSHChannelPool.java:111)
	at com.github.robtimus.filesystems.sftp.SSHChannelPool$Channel.<init>(SSHChannelPool.java:106)
	at com.github.robtimus.filesystems.sftp.SSHChannelPool.lambda$new$0(SSHChannelPool.java:75)
	at com.github.robtimus.pool.Pool.createObject(Pool.java:382)
	at com.github.robtimus.pool.Pool.fillPool(Pool.java:100)
	at com.github.robtimus.pool.Pool.<init>(Pool.java:83)
	at com.github.robtimus.filesystems.sftp.SSHChannelPool.<init>(SSHChannelPool.java:75)
	at com.github.robtimus.filesystems.sftp.SFTPFileSystem.<init>(SFTPFileSystem.java:107)
	at com.github.robtimus.filesystems.sftp.SFTPFileSystemProvider.createFileSystem(SFTPFileSystemProvider.java:76)
	at com.github.robtimus.filesystems.FileSystemMap.createFileSystem(FileSystemMap.java:127)
	at com.github.robtimus.filesystems.FileSystemMap.add(FileSystemMap.java:108)
	at com.github.robtimus.filesystems.sftp.SFTPFileSystemProvider.newFileSystem(SFTPFileSystemProvider.java:99)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:339)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:288)
	at ...
Caused by: com.jcraft.jsch.JSchAlgoNegoFailException: Algorithm negotiation fail: algorithmName="server_host_key" jschProposal="rsa-sha2-512,rsa-sha2-256,ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521" serverProposal="ssh-rsa"
	at com.jcraft.jsch.KeyExchange.guess(KeyExchange.java:147)
	at com.jcraft.jsch.Session.receive_kexinit(Session.java:602)
	at com.jcraft.jsch.Session.connect(Session.java:334)
	at com.jcraft.jsch.Session.connect(Session.java:194)
	at com.github.robtimus.filesystems.sftp.SFTPEnvironment.connect(SFTPEnvironment.java:650)
	... 17 more
```

**Comment by tklinchik** [comment]
I found a workaround after doing a little more digging:

```
JSch.setConfig("server_host_key", JSch.getConfig("server_host_key") + ",ssh-rsa")
JSch.setConfig("PubkeyAcceptedAlgorithms", JSch.getConfig("PubkeyAcceptedAlgorithms") + ",ssh-rsa")
```

**Comment by robtimus** [comment]
This is caused by the upgrade in JSch library in [version 3.0](https://github.com/robtimus/sftp-fs/releases/3.0/). The release links to  https://github.com/mwiede/jsch which already mentions this at [Why do ssh-rsa type keys not work with this JSch fork and my server?](https://github.com/mwiede/jsch#why-do-ssh-rsa-type-keys-not-work-with-this-jsch-fork-and-my-server).

You've now set the setting globally, which is fine. It can also be set per SFTP file system using `SFTPEnvironment.withConfig`:

```
SFTPEnvironment env = new SFTPEnvironment()
        // other configuration as usual
        .withConfig("server_host_key", JSch.getConfig("server_host_key") + ",ssh-rsa")
        .withConfig("PubkeyAcceptedAlgorithms", JSch.getConfig("PubkeyAcceptedAlgorithms") + ",ssh-rsa");
```

I don't plan on making this easier right now because the fix easy enough.

**Comment by robtimus** [comment]
FYI, I've just released version 3.2 that makes it a bit easier:
```
SFTPEnvironment env = new SFTPEnvironment()
        // other configuration as usual
        .withAppendedConfig("server_host_key", "ssh-rsa")
        .withAppendedConfig("PubkeyAcceptedAlgorithms", "ssh-rsa");
```
